### ISSUE TITLE
HTTP response not sent if use Vite proxy

### ISSUE THREAD

**Issue by hongquan** [oven-sh/bun#4648]
### What version of Bun is running?

1.0.0

### What platform is your computer?

Linux 6.2.0-32-generic x86_64 x86_64

### What steps can reproduce the bug?

- Create a Vite + VueJS project.
- Configure [Vite proxy](https://vitejs.dev/config/server-options.html#server-proxy), to proxy the API request to API server.
- Run Vite with bun (`bunx --bun vite`) (follow [this tutorial](https://bun.sh/guides/ecosystem/vite))
- Access a page where it makes request to API

Unfortunately, I cannot have a snippet to demonstrate. I however have an open source codebase, https://github.com/hongquan/QuanWeb, that has API backend and [VueJS frontend](https://github.com/hongquan/QuanWeb/tree/main/admin) to reproduce this issue.

### What is the expected behavior?

- The frontend gets a complete response from API.

### What do you see instead?

- The request hangs, waiting for response body. Still receive response header.

**Comment by pegiadise** [comment]
Confirming on `System Version: macOS 13.5.2 (22G91)`

**Comment by lafllamme** [comment]
any updates on this?

**Comment by LucyEgan** [comment]
This still happens on 1.0.13
And here is the stack I get during proxying the request

```
 ERROR  2:44:45 PM [vite] http proxy error at http://127.0.0.1:3000/__internal_internalpolyfillv2:                                                                                                                                                                                                        14:44:45
Error: socket hang up
    at connResetException (node:internal/errors:720:14)
    at Socket.socketOnEnd (node:_http_client:525:23)
    at Socket.emit (node:events:526:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
```

The vite process doesn't crash any more, but instead it fails on the connection to the backend and returns 500 to the client instead of the proxied request.

**Comment by hongquan** [comment]
This has been fixed by Bun v1.0.14.
